src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +583,208 @@ subroutine DustEmission (bounds, &
+         !          friction velocity (currently a constant)
+
+         if (lnd_frc_mbl(p) > 0.0_r8  .AND. tlai_lu(l)<=1_r8) then
+         !if (lnd_frc_mbl(p) > 0.0_r8  .AND. ttlai(p)<=1_r8) then
+            ! vegetation drag partition equation following Gregory Okin (2008) + Caroline Pierre et al. (2014)
+            !lai(p) = tlai_lu(l)+0.1_r8       ! LAI+SAI averaged to landunit level; the equation is undefined at lai=0, and LAI in CTSM has some zeros over deserts, so we add in a small number.
+            !lai(p) = ttlai(p) + lai0_Okin     ! ttlai = tlai+tsai. Okin-Pierre's equation is undefined at lai=0, and LAI in CTSM has some zeros over deserts, so we add in a small number. On 26 Feb 2024, dmleung changed from tlai_lu(l) to ttlai(p)
+            !if (lai(p) > 1_r8) then
+            !   lai(p)  = 1_r8   ! setting LAI = 1 to be a max value (since K_length goes to negative when LAI>1)
+            !end if              ! 
+
+            !if (ttlai(p) + vai0_Okin <= 1_r8) then
+            !   vai_Okin(p) = ttlai(p) + vai0_Okin     ! ttlai = vai = tlai+tsai. Okin-Pierre's equation is undefined at vai=0, and VAI in CTSM has some zeros over deserts, so we add in a small number. On 26 Feb 2024, dmleung changed from tlai_lu(l) to ttlai(p)
+            !end if                                    ! In the Okin-Pierre formulation, VAI has to be 0 < VAI <= 1.
+
+            vai_Okin(p) = tlai_lu(l)+vai0_Okin       ! LAI+SAI averaged to landunit level; the equation is undefined at lai=0, and LAI in CTSM has some zeros over deserts, so we add in a small number.
+            if (vai_Okin(p) > 1_r8) then
+               vai_Okin(p)  = 1_r8   ! setting LAI = 1 to be a max value (since K_length goes to negative when LAI>1)
             end if
 
-            ! slevis: adding liqfrac here, because related to effects from soil water
+            ! calculate Okin's shear stress ratio (SSR, which is vegetation drag partition factor) using Pierre's equation
+            K_length = 2_r8 * (1_r8/vai_Okin(p) - 1_r8)   ! Here LAI has to be non-zero to avoid blowup, and < 1 to avoid -ve K_length. See this equation in Leung et al. (2023). This line is Okin's formulation
+            ssr(p) = (K_length+f_0*c_e)/(K_length+c_e) ! see this equation in Caroline Pierre et al. (2014) or Leung et al. (2023). This line is Pierre's formulation.
+
+            ! calculation of the hybrid/total drag partition effect considering both rock and vegetation drag partitioning using LUH2 bare and veg fractions within a grid
+            if (lun%itype(l) == istsoil .or. lun%itype(l) == istcrop) then
+               if (patch%itype(p) == noveg) then ! if bare, uses rock drag partition factor
+                  if (dpfct_rock(p) /= dpfct_rock(p)) then ! dmleung added 24 May 2024: dpfct_rock(p) could be NaN; CLM could run when DEBUG=FALSE in env_build.xml but dies when DEBUG=TRUE (usually when checking if wnd_frc_slt > wnd_frc_thr_slt_it and if numer/denom < 30._r8 below)
```

Comment:
We should change this to use the shr_infnan_isnan function to make sure this works for all compilers and optimization levels. I can do that though when the time comes.

Reply:
Oops, I'm not so sure how to use shr_infnan_isnan. I guess I will let you do it, or I can take a look at its use in other modules later.

Reply:
Still something to do.

Reply:
I added a unit test around this, and found that in DEBUG=TRUE mode, the above check is NOT sufficient for the current intel compiler. This probably works for DEBUG=FALSE, but it aborts with a floating point error with DEBUG on.

All of which underscores the need to change it to the shr_infnan_isnan function, which I'm doing now.